python/ray/serve/tests/test_backend_state.py

```diff
@@ -159,30 +149,30 @@ def test_return_existing_goal(mock_backend_state):
 
 
 def test_create_delete_single_replica(mock_backend_state):
-    backend_state, timer, mock_replicas, goal_manager = mock_backend_state
+    backend_state, timer, goal_manager = mock_backend_state
 
-    assert replica_count(backend_state) == 0
+    assert len(replicas(backend_state)) == 0
 
     b_config_1, r_config_1 = generate_configs()
     create_goal = backend_state.create_backend("tag1", b_config_1, r_config_1)
 
     # Single replica should be created.
     backend_state.update()
-    assert replica_count(backend_state) == 1
-    assert replica_count(backend_state, states=[ReplicaState.STARTING]) == 1
-    assert mock_replicas[0].started
+    assert len(replicas(backend_state)) == 1
+    assert len(replicas(backend_state, states=[ReplicaState.STARTING])) == 1
+    assert replicas(backend_state)[0]._actor.started
```

Comment:
Do we want to just keep a reference for all the tests? 
```
starting_replica = replicas(backend_state)[0]

starting_replica._actor.started
...
starting_replica._actor.set_ready()
```

Reply:
yeah that's probably cleaner. the tests already passed here though so I'm going to merge to unblock my other PR